### PR TITLE
Interactive size const

### LINKED ISSUE #36713: Create a universal constant for minimum interactive size
Lots of places in the codebase have a hardcoded value of 48x48 pixels (material) or 44x44 pixels (cupertino) in order to size hit targets in a way that's easily interactive.  We should create a single constant for these and reuse it everywhere.

See https://github.com/flutter/flutter/pull/35829/files#r305981652 for discussion and locations.

### PR DESCRIPTION
**Pull request by justinmc**
## Description

Lots of places in the codebase have a hardcoded value of 48x48 pixels (material) or 44x44 pixels (cupertino) in order to size hit targets in a way that's easily interactive.  This PR pulls these two values out as constants and reuses them in all of the places where it was previously hardcoded.

## Related Issues

Closes https://github.com/flutter/flutter/issues/36713

## Tests

Updated tests that were using an old duplicated constant.

### CODE REVIEW

**Comment by justinmc** [comment]
I ended up having to create a widgets/constants.dart file and duplicating the material constant there to avoid circular dependencies.  Is there a better way to put these constants?  Or should I rename the widgets constant to "kMinInteractiveDimensionGeneric" or something like that?

File: packages/flutter/lib/src/material/constants.dart [hunk 1]
```diff
@@ -4,14 +4,27 @@
...
+///
+/// This is used to avoid small regions that are hard for the user to interact
+/// with. It applies to both dimensions of a region, so a square of size
+/// kMinInteractiveDimension x kMinInteractiveDimension is the smallest
+/// acceptable region that should respond to gestures.
+///
+/// See also:
+///
+///   * [kMinInteractiveDimensionCupertino]
+const double kMinInteractiveDimension = 48.0;
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/constants.dart]: This is covered in the "Touch and pointer size" section of the material spec: https://material.io/design/usability/accessibility.html

Probably should provide a link to it.

File: packages/flutter/lib/src/material/icon_button.dart [hunk 2]
```diff
@@ -31,9 +32,9 @@ const double _kMinButtonSize = 48.0;
 ///
 /// Requires one of its ancestors to be a [Material] widget.
 ///
-/// The hit region of an icon button will, if possible, be at least 48.0 pixels
-/// in size, regardless of the actual [iconSize], to satisfy the [touch target
-/// size](https://material.io/guidelines/layout/metrics-keylines.html#metrics-keylines-touch-target-size)
+/// The hit region of an icon button will, if possible, be at least
+/// kMinInteractiveDimension pixels in size, regardless of the actual
+/// [iconSize], to satisfy the [touch target size](https://material.io/guidelines/layout/metrics-keylines.html#metrics-keylines-touch-target-size)
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/icon_button.dart]: Might not be the best spec link; see earlier comment

File: packages/flutter/lib/src/widgets/text_selection.dart [hunk 3]
```diff
@@ -619,11 +620,6 @@ class _TextSelectionHandleOverlay extends StatefulWidget {
   }
 }
 
-/// The minimum size that a widget should be in order to be easily interacted
-/// with by the user.
-@visibleForTesting
-const double kMinInteractiveSize = 48.0;
```
Comment (HansMuller) [inline packages/flutter/lib/src/widgets/text_selection.dart]: Unfortunately this change makes the PR backwards incompatible. It's a pretty minor one as these things go; but tests may break. We should follow the breaking change protocol for this PR.

**Review by HansMuller — APPROVED** [review]
NICE, LGTM.

File: packages/flutter/lib/src/widgets/text_selection.dart (hunk 3 above)
Comment (justinmc) [reply packages/flutter/lib/src/widgets/text_selection.dart]: True.  I'll send out that email in a few minutes.

**Comment by justinmc** [comment]
### Breaking Change Announcement Draft

[Breaking Change] Removing kMinInteractiveSize in text_selection.dart

I'm proposing to remove the constant kMinInteractiveSize from text_selection.dart.  This constant was annotated with `@visibleForTesting`, so the impact is likely to be very small.

Issue: https://github.com/flutter/flutter/issues/36713
PR: https://github.com/flutter/flutter/pull/36964

In many places around the framework, we had duplicated hardcoded values for the minimum touch target size (48 pixels for Material and 44 for Cupertino).  The PR consolidates all of those duplicate values into a few shared constants to avoid code duplication.

Anyone who may have been using kMinInteractiveSize should replace it with kMinInteractiveDimension in widgets/constants.dart, which is added in the referenced PR.  Please let me know if you have any trouble migrating and need some help!  I'm available by email and chat.

If anyone objects to this change or would like to discuss further, please comment in the linked PR.

Thanks,

Justin

**Comment by Anspagespace** [comment]
превключвател (widget.materialTapTargetSize ?? topic.materialTapTargetSize) {
       case MaterialTapTargetSize.padded:
- ограничения = const BoxConstraints (minHeight: 48.0);
+ ограничения = const BoxConstraints (minHeight: kMinInteractiveDimension);
         прекъсване;
       case MaterialTapTargetSize.shrinkWrap:
         ограничения = const BoxConstraints ();
различно - създайте / пакети / трептене / lib / src / материал / константи.
индекс 4940576bb63..08d41965a6a 100644
--- a / пакети / flutter / lib / src / material / constants.dart
+++ b / пакети / трептене / lib / src / материал / константи.дарт
@@ -4,6 +4,15 @@
 
 import 'пакет: flutter / painting.dart';
 
+ /// Минималното измерение на всеки интерактивен регион според Материал
+ /// насоки.
+ ///
+ /// Използва се, за да се избегнат малки региони, които са трудни за взаимодействие с потребителя
+ /// с. Тя се отнася и за двете измерения на даден регион, така че квадрат с размер
+ /// kMinInteractiveDimension x kMinInteractiveDimension е най-малкото
+ /// приемлив регион, който трябва да отговаря на жестове.
+ const двойна kMinInteractiveDimension = 48.0;
+
 /// Височината на компонента на лентата с инструменти на [AppBar].
 const двойна kToolbarHeight = 56.0;
 
@@ -11,7 +20,7 @@ const двойна kToolbarHeight = 56.0;
 const двойна kBottomNavigationBarHeight = 56.0;
 
 /// Височината на лентата с раздели, съдържаща текст.
-const двоен kTextTabBarHeight = 48.0;
+ const двойна kTextTabBarHeight = kMinInteractiveDimension;
 
 /// Количеството анимации за промяна на тема трябва да продължи.
 const Продължителност kThemeChangeDuration = Продължителност (милисекунди: 200);
различно - създайте / пакети / трептене / lib / src / материал / data_table.dart b / пакети / трептене / lib / src / материал / data_table.dart
индекс 60822095313..cccd97b5584 100644
--- a / пакети / flutter / lib / src / material / data_table.dart
+++ b / пакети / трептене / lib / src / материал / data_table.dart
@@ -10,6 +10,7 @@ import 'пакет: flutter / widgets.dart';
 
 импортиране „checkbox.dart“;
 импортиране на 'color.dart';
+ импортиране „constants.dart“;
 import 'debug.dart';
 импортиране на „delier.dart“;
 import 'dropdown.dart';
@@ -262,7 +263,7 @@ class DataTable разширява StateWidget {
     this.sortColumnIndex,
     this.sortAscending = true,
     this.onSelectAll,
- this.dataRowHeight = 48.0,
+ this.dataRowHeight = kMinInteractiveDimension,
     this.headingRowHeight = 56.0,
     this.horizontalMargin = 24.0,
     this.columnSpacing = 56.0,
@@ -321,7 +322,8 @@ class DataTable разширява StateWidget {
 
   /// Височината на всеки ред (с изключение на реда, който съдържа заглавия на колоните).
   ///
- /// Тази стойност по подразбиране е 48.0, за да се придържа към спецификациите на Material Design.
+ /// Тази стойност по подразбиране е kMinInteractiveDimension, за да се придържа към материала
+ /// Спецификации на дизайна.
   крайни двойни данниRowHeight;

**Pull request merged**